src/controllers/participant.controller.ts

```diff
@@ -1,264 +1,256 @@
-            { acceptanceStatus: StatusEnum.EMAILED },
-            { phone }
-        );
-        await dbAddNewLog({
-            adminPhone: req.user?.phone as string,
-            adminId: req.user?._id as string,
-            action: "update",
-            participantId: updatedParticipant._id as string,
-        });
-        res.status(200).json({ status: "success", data: updatedParticipant });
-    } catch (error: ErrorWithStatusCode | any) {
-        res.status(error.statusCode || 500).json({
-            status: "failure",
-            data: error.message,
-        });
-    }
+	/**
+	 * #swagger.tags = ['Participants']
+	 * #swagger.description = 'Endpoint to get send an email to a participant from database'
+	 */
+	try {
+		const { phone }: { phone: string } = req.body;
+		//TODO: send email using nodemailer here
+		const originalParticipant = await dbGetParticipantByPhone(phone);
+		const updatedParticipant = await dbUpdateParticipant({ acceptanceStatus: StatusEnum.EMAILED }, { phone });
+		const diff = diffObjects(originalParticipant, updatedParticipant);
+		await dbAddNewParticipantLog({
+			initiator: req.user?._id as string,
+			action: "update",
+			target: updatedParticipant._id as string,
+			previousState: diff.OLD as Partial<ParticipantType>,
+			newState: diff.NEW as Partial<ParticipantType>,
+		});
+		res.status(200).json({ status: "success", data: updatedParticipant });
+	} catch (error: ErrorWithStatusCode | any) {
+		res.status(error.statusCode || 500).json({
+			status: "failure",
+			data: error.message,
+		});
+	}
 }
 
 //------------------Status------------------//
 
 export async function updateParticipantStatus(req: Request, res: Response) {
-  /**
-   * #swagger.tags = ['Participants']
-   * #swagger.description = 'Endpoint to update a participant's status'
-   */
-  try {
-    const { status, phone }: { status: StatusEnum, phone: string } = req.body;
-    const updatedParticipant = await dbUpdateParticipant(
-      { acceptanceStatus: status },
-      { phone }
-    );
-    await dbAddNewLog({
-      adminPhone: req.user?.phone as string,
-      adminId: req.user?._id as string,
-      action: "update",
-      participantId: updatedParticipant._id as string,
-    });
-    res.status(200).json({ status: "success", data: { status: updatedParticipant.acceptanceStatus } });
-  } catch (error: ErrorWithStatusCode | any) {
-    res.status(error.statusCode || 500).json({
-      status: "failure",
-      data: error.message,
-    });
-  }
+	/**
+	 * #swagger.tags = ['Participants']
+	 * #swagger.description = 'Endpoint to update a participant's status'
+	 */
+	try {
+		const { status, phone }: { status: StatusEnum; phone: string } = req.body;
+		const originalParticipant = await dbGetParticipantByPhone(phone);
+		const updatedParticipant = await dbUpdateParticipant({ acceptanceStatus: status }, { phone });
+		const diff = diffObjects(originalParticipant, updatedParticipant);
+		await dbAddNewParticipantLog({
```

Comment:
This block of code was repeated multiple times. It could be time to move this to a separate function.